File: mobile/modules/xmu-cookie/android/src/main/java/expo/modules/xmucookie/XmuCookieModule.kt
package expo.modules.xmucookie

import android.webkit.CookieManager
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class XmuCookieModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("XmuCookie")

    OnCreate {
      CookieManager.getInstance().setAcceptCookie(true)
    }

    AsyncFunction("getCookieForUrlAsync") { url: String ->
      CookieManager.getInstance().getCookie(url)
    }

    AsyncFunction("clearCookiesAsync") {
      val cm = CookieManager.getInstance()
      cm.removeAllCookies(null)
      cm.flush()
    }
  }
}

File: mobile/modules/xmu-cookie/src/XmuCookieModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class XmuCookieModule extends NativeModule<{}> {
  getCookieForUrlAsync(url: string): Promise<string | null>;
  clearCookiesAsync(): Promise<void>;
}

export default requireNativeModule<XmuCookieModule>('XmuCookie');
